packages/connect-react/src/components/login/LoginPasskeyScreen.tsx

```diff
@@ -0,0 +1,73 @@
+import { PasskeyChallengeCancelledError } from '@corbado/web-core';
+import log from 'loglevel';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import useLoginProcess from '../../hooks/useLoginProcess';
+import useShared from '../../hooks/useShared';
+import { LoginScreenType } from '../../types/screenTypes';
+import { LinkButton } from '../shared/LinkButton';
+import { PasskeyButton } from '../shared/PasskeyButton';
+
+export const LoginPasskeyScreen = () => {
+  const { config, navigateToScreen, setCurrentIdentifier, currentIdentifier } = useLoginProcess();
+  const { getConnectService } = useShared();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const lastLogin = getConnectService().getLastLogin();
+
+    if (!lastLogin?.identifierValue) {
+      handleFallback();
+      return;
+    }
+
+    setCurrentIdentifier(lastLogin.identifierValue);
+  }, [getConnectService]);
+
+  const handleSubmit = useCallback(async () => {
+    setLoading(true);
+
+    const res = await getConnectService().login(currentIdentifier);
+    if (res.err) {
+      setLoading(false);
+      if (res.val.ignore) {
+        return;
+      }
+
+      if (res.val instanceof PasskeyChallengeCancelledError) {
+        navigateToScreen(LoginScreenType.ErrorHard);
+        return;
+      }
+
+      log.debug('login not allowed');
+      handleFallback();
+
+      return;
+    }
+
+    setLoading(false);
+    config.onComplete(res.val.session);
+  }, [getConnectService, config, currentIdentifier]);
+
+  const handleFallback = useCallback(() => {
```

Comment:
Maybe we choose another name than "handleFallback" here? Currently, "fallback" describes the situation when passkey login will be made unavailable to the user and she has to provide a password instead (in the case of PasskeyButton this is not the case though because we handle a "fallback" differently, right?)